rfcs/0009-code-review.md

```diff
@@ -0,0 +1,173 @@
+   faster to review.
+
+
+### Principles for reviewers
+ - Try to comment on the PR in as short an interval is possible (<24
+   business hours preferably).
+   - The comment doesn't have to be the review, but can just be a
+     comment letting the reviewee know when you'll be able to review.
+     This gives them the opportunity to find another reviewer if needed.
+   - If you’re not sure how someone wants their code reviewed or how they
+     prefer to receive feedback, talk to them beforehand.
+ - Point out when something is particularly good or clever.
+   - Reviews often focus on the negative so highlight the good as well
+     when you see it.
+ - Be explicit.
+   - Include screenshots or gifs if they help make your point.
+ - Programming is subjective.
+   - Different approaches are valid.
+   - Ask for clarifications if you need any.
+ - Communicate which ideas you feel strongly about and those you don't.
+
+
+### Technical Considerations
+
+#### Logic
+ - Does the code work?
+   - Does it perform its intended function?
+   - Is the logic correct?
+   - Is the proposed UI accessible?
+ - Does this change introduce any security concerns?
+   - Does it accept or handle user data?
+ - Is all the code easily understood?
+   - Which parts are confusing to you and why?
+   - If you are taking a lot of time to understand the code, perhaps it
+     needs comments or refactoring.
+ - Can the code be simplified?
+   - Write the code you need, not the code you may need.
+ - Does similar functionality already exist in the codebase?
+   - If so, why isn't this functionality reused?
+     (Note: it's okay to not reuse -— but it's worth asking the question.)
+
+#### Comments
+ - Has all commented-out code been deleted?
+   - We can find past code using version control if needed.
+ - Are all functions commented?
+ - Do comments exist to describe the intent of any thorny logic?
+ - Are there comments that no longer make sense?
+
+#### Errors
+ - Can you think of any use case in which the code does not behave as
+   intended?
+ - Are potential errors being caught and logged?
+ - Are third-party utilities used which may return errors?
+ - Is any unusual behavior or edge-case handling described?
+
+#### Tests
+ - Do tests exist, and are they comprehensive, covering good input, bad
+   input and errors?
+ - Are tests checking at the right level?
+ - Do the existing set of tests pass?
+
+#### Documentation
+ - Does this change require an update to the README or configuration
+   files?
+ - Is the reason for the change outlined in the commit message?
+
+
+## Drawbacks
+Making the code review process too restrictive makes it more of a core
+then it should be. We want the process to be fairly light, but still
+maintain value.
+
+
+## Alternatives
+Without the guide rails the process can continue as is with different
+styles of code review being provided by different folks. This, while
+functional, can be difficult if you need different reviewers for things
```

Comment:
I think this sentence might be missing a couple words or commas? I'm finding it difficult to read and understand what it's trying to say 🤔

Reply:
Hah, missed the forest because of the trees. Better?